src/librustc_trans/trans/context.rs

```diff
@@ -856,6 +855,29 @@ impl<'b, 'tcx> CrateContext<'b, 'tcx> {
             codegen_items.insert(cgi, TransItemState::NotPredictedButGenerated);
         }
     }
+
+    /// Given the node-id of some local item that has no type
+    /// parameters, make a suitable "empty substs" for it.
+    pub fn empty_substs_for_node_id(&self, item_node_id: ast::NodeId)
```

Comment:
Heh, so they do.

On Fri, Mar 18, 2016 at 8:59 PM, Eduard-Mihai Burtescu <
notifications@github.com> wrote:

> In src/librustc_trans/trans/context.rs
> <https://github.com/rust-lang/rust/pull/32346#discussion_r56740326>:
>
> > @@ -856,6 +855,29 @@ impl<'b, 'tcx> CrateContext<'b, 'tcx> {
> >              codegen_items.insert(cgi, TransItemState::NotPredictedButGenerated);
> >          }
> >      }
> > +
> > +    /// Given the node-id of some local item that has no type
> > +    /// parameters, make a suitable "empty substs" for it.
> > +    pub fn empty_substs_for_node_id(&self, item_node_id: ast::NodeId)
>
> I don't think we want this function, all callers *already* compute
> local_def_id themselves.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/rust/pull/32346/files/378743fd079475246589c1d4ddb7b614ab58f9cc..d8c2e42975cc77263235ea29e5ed10bbaa9bcebc#r56740326>
>